src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala

```diff
@@ -0,0 +1,56 @@
+package is.hail.expr.ir.functions
+
+import is.hail.annotations.{Region, StagedRegionValueBuilder}
+import is.hail.asm4s
+import is.hail.asm4s.{Code, MethodBuilder}
+import is.hail.expr.ir._
+import is.hail.expr.{Apply, ir, types}
+import is.hail.expr.types.{TBoolean, TBooleanOptional, TInterval, TStruct}
+import is.hail.utils._
+
+object IntervalFunctions extends RegistryFunctions {
+
+  def registerAll(): Unit = {
+
+    registerCode("Interval", Array(tv("T"), tv("T"), TBoolean(), TBoolean()), TInterval(tv("T")), isDet = true) {
+      (mb, args) =>
+        val srvb = new StagedRegionValueBuilder(mb, TInterval(tv("T").t))
+        Code(
+          srvb.start(),
+          srvb.addIRIntermediate(tv("T").t)(args(0)),
+          srvb.advance(),
+          srvb.addIRIntermediate(tv("T").t)(args(1)),
+          srvb.advance(),
+          srvb.addBoolean(args(2).asInstanceOf[Code[Boolean]]),
+          srvb.advance(),
+          srvb.addBoolean(args(3).asInstanceOf[Code[Boolean]]),
+          srvb.advance(),
+          srvb.offset
```

Comment:
Is there a preference for `srvb.offset` vs `srvb.end()`

Reply:
I've been using `offset` everywhere. I've no idea if other people have preferences.